### ISSUE TITLE
REPL command history

### ISSUE THREAD

**Issue by antoinechampion** [filipw/dotnet-script#554]
I see that REPL command history has been added through [this pull request](https://github.com/filipw/dotnet-script/pull/394). However, when launching a script with `dotnet-script -i myscript`.
I am not able to view the history with up and down arrows, and I can't even use the left arrow to navigate in my current command line. My terminal is `xterm-256color`.

**Comment by filipw** [comment]
does it work in default terminal?
terminala with output/input redirects are not supported

**Comment by jasonkarns** [comment]
I have the same issue. Fails in Terminal.app without any io redirection. (up/down arrows do nothing)

**Comment by leo-labs** [comment]
It seems that #417 broke command history.

**Comment by matthewblott** [comment]
I'm guessing this is still outstanding, is it also why you can't move the cursor backwards and have to delete any characters if you want to go back along the line you're working on?

**Comment by sandyarmstrong** [comment]
> It seems that #417 broke command history.

Agreed. Reverting this change fixes history and left/right cursor movement.

I'm using bash and iTerm.app on macOS 11.6.1.

**Comment by geekley** [comment]
Same issue on Konsole, on Kubuntu 23.04.
